vmm/src/device_manager/mmio.rs

```diff
@@ -81,6 +81,7 @@ pub struct MMIODeviceManager {
     irq: u32,
     last_irq: u32,
     id_to_dev_info: HashMap<(DeviceType, String), MMIODeviceInfo>,
+    pub stdio_serial: Option<Arc<Mutex<devices::legacy::Serial>>>,
```

Comment:
nit: remove `pub` and add a getter method (this is not set outside its implementation so setter not required)

Reply:
I think we should also mark this `#[cfg(target_arch = "aarch64")]`. Even if it's just an `Option` that will stay `None` on `x86_64`, it might be helpful for anyone looking at the definition of `MMIODeviceManager` to clearly see that the serial is a MMIO-device only on `aarch64`.

Reply:
@sandreim I am not sure this was necessary.
There is a `get_device` function in this structure:
```
    /// Gets the the specified device.
    pub fn get_device(
        &self,
        device_type: DeviceType,
        device_id: &str,
    ) -> Option<&Mutex<BusDevice>> {
```
that you could use. Something of the kind:
```
device_manager
                .get_device(DeviceType::Serial, "uart")
```
Doesn t this do the job?